### PR TITLE
Closes #105: Made solrQuery shorter

### LINKED ISSUE #105: Adding many new metadata fields causes very long query strings in search
In result it may cause exceeded query size limit (default to 8kb)
It seems that this query string may be shortened (alternative is to bump up the limit)

Example query sent to solr (when searching for all results in root dataverse):
```
q=*&sort=dateSort+desc&hl=true&hl.snippets=1&hl.simple.pre=<span+class%3D"search-term-match">&hl.simple.post=</span>
&hl.fl=directionOfDetectorY-Axis2&hl.fl=imageFormat&hl.fl=directionOfDetectorY-Axis1&hl.fl=redundancyHrs&hl.fl=directionOfDetectorY-Axis3&hl.fl=resolution.Temporal&hl.fl=detectorType&hl.fl=distributionDate&hl.fl=contributorName&hl.fl=userID&hl.fl=resolution&hl.fl=externalDatabaseId&hl.fl=rMerge&hl.fl=pdbEntityId&hl.fl=coverage.Temporal&hl.fl=datasetContactAffiliation&hl.fl=journalArticleType&hl.fl=datasetLevelErrorNotes&hl.fl=state&hl.fl=citationJournal&hl.fl=otherIdValue&hl.fl=dateOfCollectionEnd&hl.fl=studyAssayOrganism
...&(a lot more of hl.fl params)...
&fl=*,score&qt=/select&facet=true&facet.query=*
&fq=dvObjectType:(dataverses+OR+datasets)&facet.field=dvCategory&facet.field=metadataSource&facet.field=publicationDate&facet.field=publicationStatus
&facet.field=authorName_ss&facet.field=subject_ss&facet.field=keywordValue_ss&facet.field=dateOfDeposit_s&facet.field=fileTypeGroupFacet
&facet.field=dvObjectType&facet.field=fileTag&facet.field=fileAccess&start=0&rows=10
```

### PR DESCRIPTION
**Pull request by dbojanek**

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchDynamicFieldPrefix.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+// -------------------- GETTERS --------------------
+    public String getPrefix() {
+        return prefix;
+    }
+
+// -------------------- LOGIC --------------------
+    public static boolean contains(String value) {
+        for (SearchDynamicFieldPrefix prefix : SearchDynamicFieldPrefix.values()) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchDynamicFieldPrefix.java]: maybe `stream.anyMatch(...)`?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java [hunk 2]
```diff
@@ -832,6 +832,24 @@ private String getNonDatasetFieldFacetLabelName(String facetLabelName, String fo
         return facetLabelName;
     }
 
+    private SolrQuery addHighlightFields(SolrQuery solrQuery, Map<String, String> solrFieldsToHightlightOnMap) {
+        Set<String> dynamicDatasetFieldsPrefixes = new HashSet<>();
+
+        for(String field : solrFieldsToHightlightOnMap.keySet()) {
+            if(field.length() >= 8 && SearchDynamicFieldPrefix.contains(field.substring(0, 8))) {
```
Comment (MrK191) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java]: won't there be a conflict if 2 fields have the same first 8 letters?

**Review by MrK191 — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java (hunk 2 above)
Comment (dbojanek) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java]: First 8 letters identify whether the field is dynamic or not (dynamic fields starts with 'dsf_xxx_' ). These fields we mask in solor by adding asterisk sign '*' so we don't have to pass them all in a query - thus making the query much shorter.

Non dynamic fields (not starting with 'dsf_xxx_') are passed to the query without any masking.

So fields: dsf_txt_name and dsf_txt_surname will be masked as single dsf_txt_* and fields dvName or dvContributorName will be passed as is.

Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java]: Hmm ok, then i propose extracting method to give more meaning like `if(isFieldDynamic)` so then you can even google what's dynamic fields in google.

**Review by MrK191 — APPROVED** [review]

**Pull request merged**